src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/zip/ZipStepExecution.java

```diff
@@ -112,16 +112,19 @@ protected Void run() throws Exception {
     static class ZipItFileCallable extends MasterToSlaveFileCallable<Integer> {
         final FilePath zipFile;
         final String glob;
+        private final String zipFileName;
 
-        public ZipItFileCallable(FilePath zipFile, String glob) {
+        public ZipItFileCallable(FilePath zipFile, String glob, String zipFileName) {
             this.zipFile = zipFile;
             this.glob = StringUtils.isBlank(glob) ? "**/*" : glob;
+            this.zipFileName = zipFileName;
```

Comment:
`zipFile.getName()` wouldn't return the correct value. If you `zip` to `test/archive.zip`, then  `zipFile.getName()` is `archive` and is not excluded correctly from the `FileSet`.